lib/src/react_client/synthetic_event_wrappers.dart

```diff
@@ -4,153 +4,634 @@
+  ///
+  /// The value must be `none`, `copy`, `link` or `move`.
+  ///
+  /// See <https://developer.mozilla.org/en-US/docs/Web/API/DataTransfer/dropEffect>
   external String get dropEffect;
+
+  /// Provides all of the types of operations that are possible.
+  ///
+  /// Must be one of `none`, `copy`, `copyLink`, `copyMove`, `link`, `linkMove`, `move`, `all` or `uninitialized`.
+  ///
+  /// See <https://developer.mozilla.org/en-US/docs/Web/API/DataTransfer/effectAllowed>
   external String get effectAllowed;
+
+  /// Contains a list of all the local files available on the data transfer.
+  ///
+  /// If the drag operation doesn't involve dragging files, this property is an empty list.
+  ///
+  /// See <https://developer.mozilla.org/en-US/docs/Web/API/DataTransfer/files>
   external List<File> get files;
+
+  /// Gives the formats that were set in the `dragstart` event.
+  ///
+  /// See <https://developer.mozilla.org/en-US/docs/Web/API/DataTransfer/types>
   external List<String> get types;
 }
 
+/// A [SyntheticEvent] wrapper that is specifically backed by a [MouseEvent].
+///
+/// See <https://developer.mozilla.org/en-US/docs/Web/API/MouseEvent>
 @JS()
-@anonymous
 class SyntheticMouseEvent extends SyntheticEvent {
+  /// [SyntheticMouseEvent]s cannot be manually instantiated.
+  ///
+  /// Use `createSyntheticMouseEvent` instead.
+  external factory SyntheticMouseEvent._();
+
+  /// Returns true if the `alt` key was down when the mouse event was fired.
+  ///
+  /// See <https://developer.mozilla.org/en-US/docs/Web/API/MouseEvent/altKey>
   external bool get altKey;
+
+  /// The button number that was pressed (if applicable) when the mouse event was fired.
+  ///
+  /// See <https://developer.mozilla.org/en-US/docs/Web/API/MouseEvent/button>
   external num get button;
+
+  /// The buttons being depressed (if any) when the mouse event was fired.
+  ///
+  /// See <https://developer.mozilla.org/en-US/docs/Web/API/MouseEvent/buttons>
   external num get buttons;
+
+  /// The X coordinate of the mouse pointer in local (DOM content) coordinates.
+  ///
+  /// See <https://developer.mozilla.org/en-US/docs/Web/API/MouseEvent/clientX>
   external num get clientX;
+
+  /// The Y coordinate of the mouse pointer in local (DOM content) coordinates.
+  ///
+  /// See <https://developer.mozilla.org/en-US/docs/Web/API/MouseEvent/clientY>
   external num get clientY;
+
+  /// Returns `true` if the `Ctrl` key was active when the key event was generated.
+  ///
+  /// See <https://developer.mozilla.org/en-US/docs/Web/API/MouseEvent/ctrlKey>
   external bool get ctrlKey;
-  external dynamic get dataTransfer;
+
+  /// The data that is transferred during a drag and drop interaction.
+  ///
+  /// See <https://developer.mozilla.org/en-US/docs/Web/API/DragEvent/dataTransfer>
+  @Deprecated('This field can return inconsistent results between dart2js and ddc. Use `wrappedDataTransfer` instead.')
+  external get dataTransfer;
```

Comment:
This property actually doesn't exist on `SyntheticMouseEvent`s in React.js. Rather, it's attached to a `SyntheticDragEvent`. Drag events look the same as mouse events, just also having this property. So actually, our mouse event here is a drag event. This also feels like something we should fix in react 6.0.0, but wasn't sure if it should be in this ticket or a followup